tests/microgrid/test_graph.py

```diff
@@ -1323,20 +1316,6 @@ def test__validate_leaf_components(self) -> None:
         ):
             graph._validate_leaf_components()
 
-        graph._graph.clear()
-        graph._graph.add_nodes_from(
-            [
-                (1, asdict(Component(1, ComponentCategory.GRID))),
-                (4, asdict(Component(4, ComponentCategory.EV_CHARGER))),
-                (5, asdict(Component(5, ComponentCategory.PV_ARRAY))),
-            ]
-        )
-        graph._graph.add_edges_from([(1, 5), (5, 4)])
-        with pytest.raises(
-            gr.InvalidGraphError, match="Leaf components with graph successors"
-        ):
-            graph._validate_leaf_components()
-
```

Comment:
I think we should replace PV_ARRAY with Battery in this case, instead of removing it.

Reply:
Ok, the test above uses the battery so we might be duplicating tests as they both validate leaf components. Anyway I can replace it with BATTERY if you think it is worth to keep it

Reply:
oh, that case is tested already.  didn't see that.  you can remove it if it is duplicated.

Reply:
I forgot that the component graph has really comprehensive tests. :-)

Reply:
Then I'll keep it as it is